### PR TITLE
Updated to latest fork-runner code

### PR DESCRIPTION
**Pull request by corruptmemory**

### CODE REVIEW

**Comment by corruptmemory** [comment]
HA! Never tested tests.  *blush*.  Let me update.

File: play-ad/src/main/scala/sbt/ad/SbtPlayAdPlugin.scala [hunk 1]
```diff
@@ -2,32 +2,38 @@ package sbt.ad
...
+import play.sbt.forkrun.{ PlayForkRun, PlayForkOptions }
+import play.forkrun.protocol.ForkConfig
+import PlayForkRun.autoImport._
+
+object SbtAdPlayPlugin extends AutoPlugin {
   import SbtAdPlugin.autoImport._
 
   object autoImport {
     val AppDynamicsPlay = config("appdynamicsplay").extend(AppDynamics)
-    val adPlayRunner = taskKey[BackgroundJobHandle]("Run play dev-mode runner with AppDynamics agent")
```
Comment (henrikengstrom) [inline play-ad/src/main/scala/sbt/ad/SbtPlayAdPlugin.scala]: Q: With this task removed how do one invoke the Play specific AD runner?

Comment (corruptmemory) [inline play-ad/src/main/scala/sbt/ad/SbtPlayAdPlugin.scala]: `appdynamicsplay:backgroundRun` should start the new fork runner: https://github.com/typesafehub/sbt-ad/pull/2/files#diff-4b85a9d0d9627f9ab87b58a4eaff8286R29

Comment (corruptmemory) [inline play-ad/src/main/scala/sbt/ad/SbtPlayAdPlugin.scala]: To wit:

```
[reactive-stocks] $ appdynamicsplay:backgroundRun
JavaOptions: List(-javaagent:/home/jim/typesafe/activator/monitoring/appdynamics/4.0.1.0/javaagent.jar, -Dappdynamics.agent.tierName=development, -Dappdynamics.agent.nodeName=activator-1424192059914, -Dappdynamics.agent.applicationName=n/a, -Dappdynamics.agent.runtime.dir=/home/jim/typesafe/activator/monitoring/appdynamics/4.0.1.0, -Dappdynamics.agent.accountName=typesafedev, -Dappdynamics.agent.accountAccessKey=836bc575396b, -Dappdynamics.controller.hostName=typesafedev.saas.appdynamics.com, -Dappdynamics.controller.port=443, -Dappdynamics.controller.ssl.enabled=true)
[info] Updating {file:/home/jim/reactive-stocks/}root...
[info] Resolving net.contentobjects.jnotify#jnotify;0.94 ...
[info] Done updating.
[success] Total time: 2 s, completed Feb 17, 2015 2:59:37 PM
[reactive-stocks] $ background log: info: Install Directory resolved to[/home/jim/typesafe/activator/monitoring/appdynamics/4.0.1.0]
background log: info: [DEBUG]: AgentInstallManager - Full Agent Registration Info Resolver is running
background log: info: [INFO]: AgentInstallManager - Full Agent Registration Info Resolver found system property [appdynamics.agent.applicationName] for application name [n/a]
background log: info: [INFO]: AgentInstallManager - Full Agent Registration Info Resolver found system property [appdynamics.agent.tierName] for tier name [development]
background log: info: [INFO]: AgentInstallManager - Full Agent Registration Info Resolver found system property [appdynamics.agent.nodeName] for node name [activator-1424192059914]
background log: info: [INFO]: AgentInstallManager - Full Agent Registration Info Resolver found system property [appdynamics.agent.runtime.dir] for agent runtime dir [/home/jim/typesafe/activator/monitoring/appdynamics/4.0.1.0]
background log: info: [INFO]: AgentInstallManager - Full Agent Registration Info Resolver using selfService [false]
background log: info: [INFO]: AgentInstallManager - Full Agent Registration Info Resolver using selfService [false]
background log: info: [INFO]: AgentInstallManager - Full Agent Registration Info Resolver using application name [n/a]
background log: info: [INFO]: AgentInstallManager - Full Agent Registration Info Resolver using tier name [development]
background log: info: [INFO]: AgentInstallManager - Full Agent Registration Info Resolver using node name [activator-1424192059914]
background log: info: [DEBUG]: AgentInstallManager - Full Agent Registration Info Resolver finished running
background log: info: [INFO]: AgentInstallManager - Agent runtime directory set to [/home/jim/typesafe/activator/monitoring/appdynamics/4.0.1.0]
background log: info: [INFO]: AgentInstallManager - Agent node directory set to [activator-1424192059914]
background log: info: [INFO]: JavaAgent - Using Java Agent Version [Server Agent v4.0.1.0 GA #2015-02-03_10-28-53 r8deac13d7ec074a610054d21bee3bf0ce8e8a2d1 31-4.0.1.next]
background log: info: [INFO]: JavaAgent - Running IBM Java Agent [No]
background log: info: [INFO]: JavaAgent - Java Agent Directory [/home/jim/typesafe/activator/monitoring/appdynamics/4.0.1.0/ver4.0.1.0]
background log: info: [INFO]: JavaAgent - Java Agent AppAgent directory [/home/jim/typesafe/activator/monitoring/appdynamics/4.0.1.0/ver4.0.1.0]
background log: info: Agent Logging Directory [/home/jim/typesafe/activator/monitoring/appdynamics/4.0.1.0/logs/activator-1424192059914]
background log: info: Running obfuscated agent
background log: info: Started AppDynamics Java Agent Successfully.
background log: info: Setting up Play fork run ... (use Ctrl+D to cancel)
background log: info: Updating {file:/home/jim/reactive-stocks/}root...
background log: info: Resolving org.scala-lang#scala-library;2.11.5 ...
background log: info: Resolving com.typesafe.play#twirl-api_2.11;1.0.2 ...
background log: info: Resolving org.apache.commons#commons-lang3;3.1 ...
background log: info: Resolving org.scala-lang.modules#scala-xml_2.11;1.0.1 ...
background log: info: Resolving com.typesafe.play#play_2.11;2.3.8 ...
background log: info: Resolving com.typesafe.play#build-link;2.3.8 ...
background log: info: Resolving com.typesafe.play#play-exceptions;2.3.8 ...
background log: info: Resolving org.javassist#javassist;3.18.2-GA ...
background log: info: Resolving com.typesafe.play#play-iteratees_2.11;2.3.8 ...
background log: info: Resolving org.scala-stm#scala-stm_2.11;0.7 ...
background log: info: Resolving com.typesafe#config;1.2.1 ...
background log: info: Resolving com.typesafe.play#play-json_2.11;2.3.8 ...
background log: info: Resolving com.typesafe.play#play-functional_2.11;2.3.8 ...
background log: info: Resolving com.typesafe.play#play-datacommons_2.11;2.3.8 ...
background log: info: Resolving joda-time#joda-time;2.3 ...
background log: info: Resolving org.joda#joda-convert;1.6 ...
background log: info: Resolving com.fasterxml.jackson.core#jackson-annotations;2.3.2 ...
background log: info: Resolving com.fasterxml.jackson.core#jackson-core;2.3.2 ...
background log: info: Resolving com.fasterxml.jackson.core#jackson-databind;2.3.2 ...
background log: info: Resolving org.scala-lang#scala-reflect;2.11.1 ...
background log: info: Resolving org.scala-lang.modules#scala-parser-combinators_2.11;1.0.1 ...
background log: info: Resolving io.netty#netty;3.9.3.Final ...
background log: info: Resolving com.typesafe.netty#netty-http-pipelining;1.1.2 ...
background log: info: Resolving org.slf4j#slf4j-api;1.7.6 ...
background log: info: Resolving org.slf4j#jul-to-slf4j;1.7.6 ...
background log: info: Resolving org.slf4j#jcl-over-slf4j;1.7.6 ...
background log: info: Resolving ch.qos.logback#logback-core;1.1.1 ...
background log: info: Resolving ch.qos.logback#logback-classic;1.1.1 ...
background log: info: Resolving commons-codec#commons-codec;1.9 ...
background log: info: Resolving xerces#xercesImpl;2.11.0 ...
background log: info: Resolving xml-apis#xml-apis;1.4.01 ...
background log: info: Resolving javax.transaction#jta;1.1 ...
background log: info: Resolving com.typesafe.play#play-ws_2.11;2.3.8 ...
background log: info: Resolving com.google.guava#guava;16.0.1 ...
background log: info: Resolving com.ning#async-http-client;1.8.15 ...
background log: info: Resolving oauth.signpost#signpost-core;1.2.1.2 ...
background log: info: Resolving oauth.signpost#signpost-commonshttp4;1.2.1.2 ...
background log: info: Resolving org.apache.httpcomponents#httpcore;4.0.1 ...
background log: info: Resolving org.apache.httpcomponents#httpclient;4.0.1 ...
background log: info: Resolving commons-logging#commons-logging;1.1.1 ...
background log: info: Resolving com.typesafe.akka#akka-actor_2.11;2.3.9 ...
background log: info: Resolving com.typesafe.akka#akka-slf4j_2.11;2.3.9 ...
background log: info: Resolving org.webjars#bootstrap;3.0.0 ...
background log: info: Resolving org.webjars#jquery;1.9.0 ...
background log: info: Resolving org.webjars#flot;0.8.0 ...
background log: info: Resolving org.webjars#jquery-colorhelpers;1.1 ...
background log: info: Resolving org.webjars#explorercanvas;r3 ...
background log: info: Resolving com.typesafe.play#play-test_2.11;2.3.8 ...
background log: info: Resolving junit#junit;4.11 ...
background log: info: Resolving org.hamcrest#hamcrest-core;1.3 ...
background log: info: Resolving org.specs2#specs2-core_2.11;2.3.12 ...
background log: info: Resolving org.specs2#specs2-matcher_2.11;2.3.12 ...
background log: info: Resolving org.specs2#specs2-common_2.11;2.3.12 ...
background log: info: Resolving org.scalaz#scalaz-core_2.11;7.0.6 ...
background log: info: Resolving org.scalaz#scalaz-concurrent_2.11;7.0.6 ...
background log: info: Resolving org.scalaz#scalaz-effect_2.11;7.0.6 ...
background log: info: Resolving org.specs2#specs2-junit_2.11;2.3.12 ...
background log: info: Resolving org.specs2#specs2-mock_2.11;2.3.12 ...
background log: info: Resolving org.mockito#mockito-core;1.9.5 ...
background log: info: Resolving org.objenesis#objenesis;1.0 ...
background log: info: Resolving org.specs2#specs2-matcher-extra_2.11;2.3.12 ...
background log: info: Resolving org.specs2#specs2-analysis_2.11;2.3.12 ...
background log: info: Resolving org.scala-lang#scala-compiler;2.11.0 ...
background log: info: Resolving org.specs2#classycle;1.4.3 ...
background log: info: Resolving com.novocode#junit-interface;0.11-RC1 ...
background log: info: Resolving org.scala-sbt#test-interface;1.0 ...
background log: info: Resolving com.google.code.findbugs#jsr305;2.0.3 ...
background log: info: Resolving org.fluentlenium#fluentlenium-festassert;0.9.2 ...
background log: info: Resolving org.fluentlenium#fluentlenium-core;0.9.2 ...
background log: info: Resolving org.seleniumhq.selenium#selenium-java;2.39.0 ...
background log: info: Resolving org.seleniumhq.selenium#selenium-android-driver;2.39.0 ...
background log: info: Resolving org.seleniumhq.selenium#selenium-remote-driver;2.39.0 ...
background log: info: Resolving cglib#cglib-nodep;2.1_3 ...
background log: info: Resolving org.json#json;20080701 ...
background log: info: Resolving org.seleniumhq.selenium#selenium-api;2.39.0 ...
background log: info: Resolving org.apache.httpcomponents#httpclient;4.3.1 ...
background log: info: Resolving org.apache.httpcomponents#httpcore;4.3 ...
background log: info: Resolving commons-logging#commons-logging;1.1.3 ...
background log: info: Resolving org.apache.commons#commons-exec;1.1 ...
background log: info: Resolving net.java.dev.jna#jna;3.4.0 ...
background log: info: Resolving net.java.dev.jna#platform;3.4.0 ...
background log: info: Resolving org.seleniumhq.selenium#selenium-chrome-driver;2.39.0 ...
background log: info: Resolving org.seleniumhq.selenium#selenium-htmlunit-driver;2.39.0 ...
background log: info: Resolving net.sourceforge.htmlunit#htmlunit;2.13 ...
background log: info: Resolving xalan#xalan;2.7.1 ...
background log: info: Resolving xalan#serializer;2.7.1 ...
background log: info: Resolving commons-collections#commons-collections;3.2.1 ...
background log: info: Resolving org.apache.httpcomponents#httpmime;4.3.1 ...
background log: info: Resolving net.sourceforge.htmlunit#htmlunit-core-js;2.13 ...
background log: info: Resolving net.sourceforge.nekohtml#nekohtml;1.9.19 ...
background log: info: Resolving net.sourceforge.cssparser#cssparser;0.9.11 ...
background log: info: Resolving org.w3c.css#sac;1.3 ...
background log: info: Resolving commons-io#commons-io;2.2 ...
background log: info: Resolving org.eclipse.jetty#jetty-websocket;8.1.12.v20130726 ...
background log: info: Resolving org.eclipse.jetty#jetty-util;8.1.12.v20130726 ...
background log: info: Resolving org.eclipse.jetty#jetty-io;8.1.12.v20130726 ...
background log: info: Resolving org.eclipse.jetty#jetty-http;8.1.12.v20130726 ...
background log: info: Resolving org.seleniumhq.selenium#selenium-firefox-driver;2.39.0 ...
background log: info: Resolving org.seleniumhq.selenium#selenium-ie-driver;2.39.0 ...
background log: info: Resolving org.seleniumhq.selenium#selenium-iphone-driver;2.39.0 ...
background log: info: Resolving org.seleniumhq.selenium#selenium-safari-driver;2.39.0 ...
background log: info: Resolving org.webbitserver#webbit;0.4.14 ...
background log: info: Resolving org.seleniumhq.selenium#selenium-support;2.39.0 ...
background log: info: Resolving org.easytesting#fest-assert;1.4 ...
background log: info: Resolving org.easytesting#fest-util;1.1.6 ...
background log: info: Resolving com.typesafe.akka#akka-testkit_2.11;2.3.9 ...
background log: info: Resolving com.typesafe.play#play-docs_2.11;2.3.8 ...
background log: info: Resolving org.scala-lang#scala-library;2.11.1 ...
background log: info: Resolving com.typesafe.akka#akka-actor_2.11;2.3.4 ...
background log: info: Resolving com.typesafe.akka#akka-slf4j_2.11;2.3.4 ...
background log: info: Resolving com.typesafe.play#play-doc_2.11;1.1.0 ...
background log: info: Resolving org.pegdown#pegdown;1.4.0 ...
background log: info: Resolving org.parboiled#parboiled-java;1.1.5 ...
background log: info: Resolving org.parboiled#parboiled-core;1.1.5 ...
background log: info: Resolving org.ow2.asm#asm;4.1 ...
background log: info: Resolving org.ow2.asm#asm-tree;4.1 ...
background log: info: Resolving org.ow2.asm#asm-analysis;4.1 ...
background log: info: Resolving org.ow2.asm#asm-util;4.1 ...
background log: info: Resolving commons-io#commons-io;2.4 ...
background log: info: Resolving org.scala-lang#scala-compiler;2.11.5 ...
background log: info: Resolving org.scala-lang#scala-reflect;2.11.5 ...
background log: info: Resolving org.scala-lang.modules#scala-xml_2.11;1.0.3 ...
background log: info: Resolving org.scala-lang.modules#scala-parser-combinators_2.11;1.0.3 ...
background log: info: Resolving jline#jline;2.12 ...
background log: info: Resolving com.typesafe.trace#echo-trace-akka-2.3.9_2.11;0.1.12 ...
background log: info: Resolving com.typesafe.trace#echo-trace-scala-protobuf25-2.11;0.1.12 ...
background log: info: Resolving com.typesafe.trace#echo-trace211-protobuf25_2.11;0.1.12 ...
background log: info: Resolving com.typesafe.trace#echo-event211-protobuf25_2.11;0.1.12 ...
background log: info: Resolving com.typesafe.trace#echo-trace-protocol-protobuf25;0.1.12 ...
background log: info: Resolving com.google.protobuf#protobuf-java;2.5.0 ...
background log: info: Resolving org.aspectj#aspectjrt;1.8.4 ...
background log: info: Resolving com.typesafe.akka#akka-remote_2.11;2.3.9 ...
background log: info: Resolving org.uncommons.maths#uncommons-maths;1.2.2a ...
background log: info: Resolving org.fusesource#sigar;1.6.4 ...
background log: info: Resolving log4j#log4j;1.2.15 ...
background log: info: Resolving com.typesafe.trace#echo-trace-play-2.3.8_2.11;0.1.12 ...
background log: info: Resolving com.typesafe.trace#echo-trace-play-common;0.1.12 ...
background log: info: Resolving org.aspectj#aspectjweaver;1.8.4 ...
background log: info: Resolving com.typesafe.trace#echo-sigar-libs;0.1.12 ...
background log: info: Resolving com.typesafe.play#fork-run_2.11;2.3.8 ...
background log: info: Resolving com.typesafe.play#fork-run-protocol_2.11;2.3.8 ...
background log: info: Resolving com.typesafe.play#run-support_2.11;2.3.8 ...
background log: info: Resolving com.typesafe.sbtrc#client-2-11;0.3.1 ...
background log: info: Resolving org.scala-sbt#serialization_2.11;0.1.0 ...
background log: info: Resolving org.scala-lang.modules#scala-pickling_2.11;0.10.0 ...
background log: info: Resolving org.scala-lang#scala-compiler;2.11.4 ...
background log: info: Resolving org.scala-lang.modules#scala-xml_2.11;1.0.2 ...
background log: info: Resolving org.scala-lang.modules#scala-parser-combinators_2.11;1.0.2 ...
background log: info: Resolving org.json4s#json4s-core_2.11;3.2.10 ...
background log: info: Resolving org.json4s#json4s-ast_2.11;3.2.10 ...
background log: info: Resolving com.thoughtworks.paranamer#paranamer;2.6 ...
background log: info: Resolving org.spire-math#jawn-parser_2.11;0.6.0 ...
background log: info: Resolving org.spire-math#json4s-support_2.11;0.6.0 ...
background log: info: Resolving org.scala-sbt#launcher-interface;0.13.8-M3 ...
background log: info: Resolving org.scala-sbt#interface;0.13.8-M3 ...
background log: info: Resolving org.scala-sbt#io_2.11;0.13.8-M3 ...
background log: info: Resolving org.scala-sbt#control_2.11;0.13.8-M3 ...
background log: info: Resolving org.scala-sbt#collections_2.11;0.13.8-M3 ...
background log: info: Resolving com.typesafe.sbtrc#actor-client-2-11;0.3.1 ...
background log: info: Resolving com.typesafe.akka#akka-actor_2.11;2.3.8 ...
background log: info: Resolving com.typesafe.akka#akka-testkit_2.11;2.3.8 ...
background log: info: Resolving net.contentobjects.jnotify#jnotify;0.94 ...
background log: info: Done updating.
background log: info: 
background log: info: --- (Running the application, auto-reloading is enabled) ---
background log: info: 
background log: info: play - Listening for HTTP on /0:0:0:0:0:0:0:0:9000
background log: info: 
background log: info: (Server started, use Ctrl+D to stop and go back to the console...)
background log: info: 
```

**Comment by henrikengstrom** [comment]
LGTM

**Pull request merged**